### PR TITLE
Update mechanism for handling form errors with formik

### PR DESCRIPTION
**Pull request by StKostyantin**
This pull request for update mechanism for handling form errors with `FieldError`. Also, it fixes the #871 issue.

### CODE REVIEW

File: modules/user/client-react/components/ForgotPasswordForm.web.jsx [hunk 1]
```diff
@@ -10,7 +10,8 @@ const forgotPasswordFormSchema = {
   email: [required, email]
 };
 
-const ForgotPasswordForm = ({ handleSubmit, error, sent, values, t }) => {
+const ForgotPasswordForm = ({ handleSubmit, errors, sent, values, t }) => {
+  console.log('sent', sent);
```
Comment (evgenijolshanskij) [inline modules/user/client-react/components/ForgotPasswordForm.web.jsx]: The final version of the PR shouldn't contain console.log()

File: modules/user/client-react/components/ForgotPasswordForm.web.jsx [hunk 2]
```diff
@@ -22,7 +23,7 @@ const ForgotPasswordForm = ({ handleSubmit, error, sent, values, t }) => {
         value={values.email}
       />
       <div className="text-center">
-        {error && <Alert color="error">{error}</Alert>}
+        {errors && errors.handleErr && <Alert color="error">{errors.handleErr}</Alert>}
```
Comment (evgenijolshanskij) [inline modules/user/client-react/components/ForgotPasswordForm.web.jsx]: Why `handleErr`? Is it a function?

File: modules/user/client-react/containers/ForgotPassword.jsx [hunk 3]
```diff
@@ -1,17 +1,34 @@
...
 import ForgotPasswordView from '../components/ForgotPasswordView';
 
 import FORGOT_PASSWORD from '../graphql/ForgotPassword.graphql';
 
 class ForgotPassword extends React.Component {
+  state = {
+    sent: false
+  };
+
+  onSubmit = ({ forgotPassword, t }) => async values => {
```
Comment (evgenijolshanskij) [inline modules/user/client-react/containers/ForgotPassword.jsx]: Every container component contains the same code snippet. Is there any way to take it out and omit repetition?

File: modules/core/client-react/FieldAdapter.jsx [hunk 4]
```diff
@@ -20,6 +20,7 @@ class FieldAdapter extends Component {
 
   constructor(props) {
     super(props);
+    FieldAdapter;
```
Comment (vlasenko) [inline modules/core/client-react/FieldAdapter.jsx]: Hm, what is this?

Comment (StKostyantin) [reply modules/core/client-react/FieldAdapter.jsx]: fix

File: modules/user/client-react/containers/ForgotPassword.jsx (hunk 3 above)
Comment (StKostyantin) [reply modules/user/client-react/containers/ForgotPassword.jsx]: fix

File: modules/user/client-react/components/ForgotPasswordForm.web.jsx (hunk 2 above)
Comment (StKostyantin) [reply modules/user/client-react/components/ForgotPasswordForm.web.jsx]: fix

File: modules/user/client-react/containers/UserAdd.jsx [hunk 5]
```diff
@@ -56,16 +56,7 @@ export default compose(
...
-
-          if (history) {
-            return history.push('/users/');
-          }
-          if (navigation) {
-            return navigation.goBack();
-          }
+          return addUser;
         } catch (e) {
           console.log(e.graphQLErrors);
```
Comment (vlasenko) [inline modules/user/client-react/containers/UserAdd.jsx]: Why do we leave silly error handling code like this? Maybe now we can remove try catch block?

File: modules/user/client-react/containers/Register.jsx [hunk 6]
```diff
@@ -1,44 +1,49 @@
...
-          } else if (history) {
-            history.push('/profile');
-          } else if (navigation) {
-            navigation.goBack();
-          }
+          } = await mutate({ variables: { input: { username, email, password } } });
+          return register;
         } catch (e) {
-          console.log(e.graphQLErrors);
+          throw e;
```
Comment (vlasenko) [inline modules/user/client-react/containers/Register.jsx]: Why do we do catch and rethrow? Maybe we should just remove try catch block here?

File: modules/user/client-react/containers/UserAdd.jsx (hunk 5 above)
Comment (StKostyantin) [reply modules/user/client-react/containers/UserAdd.jsx]: fix

File: modules/user/client-react/containers/Register.jsx (hunk 6 above)
Comment (StKostyantin) [reply modules/user/client-react/containers/Register.jsx]: fix

**Pull request closed without merging**